tools/eslint-rules/buffer-constructor.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @fileoverview Require use of new Buffer constructor methods in lib
+ * @author James M Snell
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+const msg = 'Use of the Buffer() constructor has been deprecated. ' +
+            'Please use either Buffer.alloc(), Buffer.allocUnsafe(), ' +
+            'or Buffer.from()';
+
+function test(context, node, which) {
+  if (node.type === which && node.callee.name === 'Buffer') {
```

Comment:
Oops, no, I'm wrong, you do mean `node.callee.name`. Not sure why I'm getting weird/surprising results with this rule, though...